native/frontend/src/org/jetbrains/kotlin/resolve/konan/diagnostics/NativeTopLevelSingletonChecker.kt

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2010-2020 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.resolve.konan.diagnostics
+
+
+import org.jetbrains.kotlin.descriptors.*
+import org.jetbrains.kotlin.name.FqName
+import org.jetbrains.kotlin.psi.KtDeclaration
+import org.jetbrains.kotlin.psi.KtProperty
+import org.jetbrains.kotlin.resolve.checkers.DeclarationChecker
+import org.jetbrains.kotlin.resolve.checkers.DeclarationCheckerContext
+import org.jetbrains.kotlin.resolve.hasBackingField
+
+object NativeTopLevelSingletonChecker : DeclarationChecker {
+    private val threadLocalFqName = FqName("kotlin.native.concurrent.ThreadLocal")
+
+    private val DeclarationDescriptor.isInsideTopLevelSingletonWithoutThreadLocal: Boolean
```

Comment:
It seems, that companion objects may pass this check

Reply:
That's ok

Reply:
Added tests for companion objects